src/main/java/io/github/classgraph/ClasspathElementDir.java

```diff
@@ -396,16 +401,19 @@ private void scanPathRecursively(final Path path, final LogNode log) {
             return;
         }
         Collections.sort(pathsInDir);
+        FileUtils.FileAttributesGetter getFileAttributes = FileUtils.createCachedAttributesGetter();
```

Comment:
Can you help me understand the purpose of the cached attributes here? As far as I see it, each path is used once with the getFileAttributes. I'm likely missing smth.

Reply:
Nice catch! I believe when I introduced the cache based lookup it was still the case that we needed it. Maybe I had a version were _'Recurse into subdirectories'_ also used the already cached attributes to check if a path is a directory and I missed it from the final state.

Reply:
Feel free to take another pass at this! Thanks!